### ISSUE TITLE
Unclear if dependencies are included

### ISSUE THREAD

**Issue by pilhuhn** [quarkusio/quarkus#23015]
**Describe the bug**

I was looking for Drools CEP on Quarkus and typed in 'drools'  in the filter field, which resulted in 5 hits.
I selected CEP. Now the others did not get marked and it is/was unclear to me if dependencies are
automatically added or not. 

**Expected behavior**

Needed dependencies are either automatically selected. Or some help text says that they will
magically appear in the pom.

**Actual behavior**

User feels unsure what will happen.

**Screenshots**

![Bildschirmfoto 2022-01-19 um 11 16 31](https://user-images.githubusercontent.com/208246/150111272-ebda9624-386f-4a2f-9d46-e83c2979a965.png)

**Environment (please complete the following information):**
 - Browser: 
 - OS: 

**Additional context**
(Add any other context about the problem here.)

**Comment by pilhuhn** [comment]
No, not mvn deps - they would get pulled in and all is good.
Here those are more logical ones. If I only select the one I had in the screenshot, the kogito-main one + resteasy + resteasy-jackson should also be added to the generated pom (from code.quarkus.io).

Theoretically, the -addon could include those in its dependencies, but then it is probably harder to get rid of them if they are not needed (like when setting kogito.generate.rest = false )

**Comment by maxandersen** [comment]
I read it as these are using json features and thus we could deduce it needs one or in case of reactive not to be added ...but don't know if the right metadata/limits are used ?

**Comment by aloubyansky** [comment]
This sounds like a case for capability requirements. Which we haven't yet implemented, although I did think it would be a matter of time when we get a request for them. It will also mean more UI work for code.quarkus - highlighting or pre-selecting (in case the requirements aren't ambiguous)  additional extension requirements for the already selected extensions by the user.

**Comment by aloubyansky** [comment]
BTW, resteasy-jackson already depends on resteasy, so simply resteasy-jackson will be enough in your case.

**Comment by maxandersen** [comment]
I don't think we need code quarkus work initially. If we just start by having it add recommended default deps and throw error when the combinations are bad. We can start doing this at command level - not require ui work upfront.

**Comment by aloubyansky** [comment]
I think it's more complicated than that. If there are alternatives to defaults, we'll have to be detecting that in case the user selects those alternatives and not automatically select the defaults that could conflict with the selected alternatives.

**Comment by kstan79** [comment]
tried, me too. no matter use cli or download .zip from code generator

**Comment by tpodg** [comment]
Would it be possible to support Kotlin serialization too? Currently it looks like only Jackson is supported, if anything else is used for json serialisation, the same missing capability exception is thrown.
